tests/test_updater_delegation_graphs.py

```diff
@@ -425,6 +426,94 @@ def test_hash_bins_graph_traversal(
         finally:
             self.teardown_subtest()
 
+    @dataclass
+    class SuccinctHashTestCase:
+        hash_prefix_len: int
+        target_path: str
+        expected_visited_bin: str
+
+    succinct_bins_graph: utils.DataSet = {
+        "case 1": SuccinctHashTestCase(
+            hash_prefix_len=1,
+            target_path="othermissingpath",
+            expected_visited_bin="delegated-0",
+        ),
+        "case 2": SuccinctHashTestCase(
+            hash_prefix_len=1,
+            target_path="missingpath",
+            expected_visited_bin="delegated-1",
+        ),
+        "case 3": SuccinctHashTestCase(
+            hash_prefix_len=2,
+            target_path="foo",
+            expected_visited_bin="delegated-0",
+        ),
+        "case 4": SuccinctHashTestCase(
+            hash_prefix_len=2,
+            target_path="doo",
+            expected_visited_bin="delegated-1",
+        ),
+        "case 5": SuccinctHashTestCase(
+            hash_prefix_len=2,
+            target_path="missingpath",
+            expected_visited_bin="delegated-2",
+        ),
+        "case 6": SuccinctHashTestCase(
+            hash_prefix_len=2,
+            target_path="bar",
+            expected_visited_bin="delegated-3",
+        ),
```

Comment:
nice test! Have you tried if it's slow with "real" numbers like if your length is 8 or 16?